### ISSUE TITLE
If I were to start Angular now, should I go with Angular 1 or 2?

### ISSUE THREAD

**Issue by kallaway** [toddmotto/ama#8]
Hi Todd,

I've been following your projects for a while, and when it comes to a question about Angular, you are the first person that comes to mind.

My Question: for someone who knows some JavaScript, but is looking to add a framework to their skill set, would you recommend starting with Angular 1 or 2? 

There are a lot of variables in this, which makes it harder to decide: the majority of tutorials I find are obviously for Angular 1, but the ones I've found on Angular 2 get too far too fast and it's hard to follow (they assume a lot of other knowledge, which I am still acquiring), and I don't know enough about both versions of a framework to pick the right one.

Thank you so much in advance!

All the Best,
Alex

**Comment by toddmotto** [comment]
Hey Alexander! Let's see...

### Angular 1.x

Angular has been around for many years now, and is highly refined and a very mature project. It's still going to be around for many years to come I believe, there are some massive applications built with it and new projects starting today being built in Angular 1.x - which is why I'd suggest learning it right now.

When you've written a Controller, bound `$scope`, written a Service, compiled templates and have a great grasp of the built-in Directives (such as `ng-click`, `ng-repeat`) then it's time in my eyes to at least start taking a lot at Angular 2. The complete rewrite with Angular 2 offers very little migration without fully rewriting or restructuring components now.

### Angular 2

Angular 2 just went into beta release, so there is still a lot of work to be done. There aren't many libraries, there aren't any StackOverflow posts, there is a community - but at the pace Angular 2 has changed we're all playing catch up. This makes it difficult to recommend learning Angular 2 right now for me at least, I'd rather learn something when it's absolutely final. That _isn't_ to say, that you shouldn't learn the concepts and look at the APIs and their progression on a weekly basis to see what's changed and the status of the project.

It will likely take a long while for Angular 2 to mature, for the community to grow and for open source projects to spawn to take us into making a transition to Angular 2 from 1.x. That being said, there will be companies using Angular 1.x for a long time due to the reliability of the framework. Big companies aren't going to place bets on a beta product right now (just because it's cool and new), there is a lot to learn and build.

### Focus

Angular 1.x has seen very rapid growth and adoption in the last few years, and it will take a long time for Angular 1.x developers to learn Angular 2 skills and map concepts and new APIs across to implementing a new application/architecture.

My best advice is to stay focused on learning the fundamental principles behind Angular, such as the built-in Directives, Services, how templates work, DOM manipulation and the programming side to managing your components - at this point Angular 2 will be easier to move to with that knowledge.

For a new Angular developer seeing `[(ng-model)]=value` and then `(click)=fn` would likely confuse them a lot. This is where your Angular 1.x experience would come into play and you'd understand that this would likely be some sort of syntax change. You'll then look up the new syntax changes, remember it and start writing Angular 2 code off the back of the 1.x knowledge. At least in my eyes this seems a sensible roadmap. There are a lot of key principles in Angular 1.x that I strongly recommend learning!

**Comment by bruinebeer** [comment]
Hi todd, i just started diving into angular, but i have been reading a lot of posts regarding react vs angular on the internet.

The consensus seems to be that angular 1 is somewhat of a horrible monster and they are trying to fix stuff in angular 2. Are they right or are they just react fanboys, not truly understanding the capabilities or focus of angular?

Although i'll likely continue to educate myself about angular, i think they're right about one thing. Angular is realy realy realy complex (i've just dived into directives, transclusion, compile...), almost like learning a new language.

What's your standpoint on react vs angular?

**Comment by m-ret** [comment]
Hey, bruinebeer. This question is not to me but I can tell you a couple things because I have over a year working with both frameworks.

I don't think about React vs Angular. I think I like JS and that I would like to learn everything about it. Just imagine Angular comes from Google, and React from Facebook, both have very good programmers, and work hard to get it done. Every framework has issues, so people telling that Angular is a mess, is because they probably haven't use React that much, as long as the app is growing, you will be checking what is missing in the frameworks and the issues they have. Also, not every framework is good for every app, depending on the architecture and depending how big is the app, you could choose one or the other. I feel very comfortable working with this 2 frameworks that I can't say which one is better, both have GREAT features. 

Sounds stupid but I love one thing in Angular: is like learning a new language but you know that JS is there. And from React: I love that it is almost pure JS. 

So the thing is: JS is beautiful, and you have to explore it. You can't underestimate Angular or React, go for it, use both, then choose your prefer one, but having in mind the kind of project you are going to be working on.

**Comment by bruinebeer** [comment]
Cool, just what i needed to know :sunglasses:
